src/responder/common/responder_common.c

```diff
@@ -1024,12 +1094,22 @@ int activate_unix_sockets(struct resp_ctx *rctx,
 
         if (ret == numfds) {
             rctx->lfd = SD_LISTEN_FDS_START;
-            ret = sd_is_socket_unix(rctx->lfd, SOCK_STREAM, 1, NULL, 0);
+
+            ret = get_socket_path_in_use(tmp_ctx, rctx->lfd, &socket_path_in_use);
+            if (ret != EOK) {
+                DEBUG(SSSDBG_MINOR_FAILURE, "Unable to obtain socket path\n");
+            }
+
+            ret = sd_is_socket_unix(rctx->lfd, SOCK_STREAM, 1, socket_path_in_use, 0);
             if (ret < 0) {
                 DEBUG(SSSDBG_CRIT_FAILURE,
                       "Activated socket is not a UNIX listening socket\n");
                 ret = EIO;
                 goto done;
+            } else if (ret > 0){
```

Comment:
`sd_is_socket_unix` returns positive value if the conditions were met, negative value on error and `0` if conditions were **not** met.